### DISCUSSION TITLE
Proof verification within a guest

### DISCUSSION THREAD

**Discussion by cf** [risc0/risc0#286] [Ideas]
Is there currently any plans to allow for porting proof verification to run in a guest?
This would be incredibly useful as it would allow for proof aggregation.

**Comment by flaub** [comment]
This is currently on the roadmap and we're actively working on it!

**Comment by flaub** [comment]
A failing test is being added: https://github.com/risc0/risc0/pull/259

Our current priority is to get this test to pass.

**Comment by sunny-g** [comment]
~Is this still on the roadmap? I've noticed that I can't import `Receipt` within a guest (gated behind `#[cfg(not(target_os = "zkvm"))]`). Original issue [here](https://github.com/risc0/risc0/issues/338)~

~edit: alternatively, regarding [this excellent talk](https://www.youtube.com/watch?v=wkIBN2CGJdc), could you take two "final" STARK proofs of the same ELF but with different inputs and aggregate them with the recursion circuit?~

Seems like it just landed, and the guest API is being worked on [here[(https://github.com/risc0/risc0/pull/992) - thanks team, this is so exciting!
